lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java

```diff
@@ -331,22 +334,79 @@ private JourneyResponse getJourneyResponse(
     private VerifiableCredentialBundle getVerifiableCredentials(
             String userId, String evcsAccessToken)
             throws CredentialParseException, EvcsServiceException {
-        if (configService.enabled(EVCS_READ_ENABLED)) {
+
+        var tacticalVcs = verifiableCredentialService.getVcs(userId);
+
+        if (configService.enabled(EVCS_WRITE_ENABLED) || configService.enabled(EVCS_READ_ENABLED)) {
             var vcs =
                     evcsService.getVerifiableCredentialsByState(
-                            userId, evcsAccessToken, CURRENT, PENDING_RETURN);
-            var pendingReturnVcs = vcs.get(PENDING_RETURN);
-            // use pending return vcs to determine identity if available
-            if (!isNullOrEmpty(pendingReturnVcs)) {
-                return new VerifiableCredentialBundle(pendingReturnVcs, true, true);
-            }
-            var currentVcs = vcs.get(CURRENT);
-            if (!isNullOrEmpty(currentVcs)) {
-                return new VerifiableCredentialBundle(currentVcs, true, false);
+                            userId, evcsAccessToken, CURRENT, PENDING_RETURN, PENDING);
+
+            logIdentityMismatches(tacticalVcs, vcs);
+
+            if (configService.enabled(EVCS_READ_ENABLED)) {
+                var pendingReturnVcs = vcs.get(PENDING_RETURN);
+                // use pending return vcs to determine identity if available
+                if (!isNullOrEmpty(pendingReturnVcs)) {
+                    return new VerifiableCredentialBundle(pendingReturnVcs, true, true);
+                }
+                var currentVcs = vcs.get(CURRENT);
+                if (!isNullOrEmpty(currentVcs)) {
+                    return new VerifiableCredentialBundle(currentVcs, true, false);
+                }
             }
         }
-        return new VerifiableCredentialBundle(
-                verifiableCredentialService.getVcs(userId), false, false);
+        return new VerifiableCredentialBundle(tacticalVcs, false, false);
+    }
+
+    @ExcludeFromGeneratedCoverageReport
+    private void logIdentityMismatches(
+            List<VerifiableCredential> tacticalVcs,
+            Map<EvcsVCState, List<VerifiableCredential>> evcsEvcs) {
+
+        var migratedTacticalVcStrings =
+                tacticalVcs.stream()
+                        .filter(vc -> vc.getMigrated() != null)
+                        .map(VerifiableCredential::getVcString)
+                        .collect(Collectors.toSet());
+
+        // if we have pending vcs just check those
+        var evcsToCheck = evcsEvcs.get(PENDING);
+        if (isNullOrEmpty(evcsToCheck)) {
+            evcsToCheck = evcsEvcs.get(CURRENT);
+        }
+        var allTacticalVcStrings =
+                tacticalVcs.stream()
+                        .map(VerifiableCredential::getVcString)
+                        .collect(Collectors.toSet());
+        var evcsVcStrings =
+                isNullOrEmpty(evcsToCheck)
```

Comment:
Minor but it might be neater to sort out `evcsToCheck` and avoid this null check. you can use `Optional` if you want - although it's obviously better:
```
var evcsToCheck = Optional.of(vcs.get(PENDING_RETURN))
        .or(() -> Optional.of(vcs.get(CURRENT)))
        .orElse(List.of());
```
(Java monads are all pretty rubbish 😞 )